lite2/client.go

```diff
@@ -520,12 +523,14 @@ func (c *Client) VerifyHeader(newHeader *types.SignedHeader, newVals *types.Vali
 		return errors.Errorf("header at more recent height #%d exists", c.trustedHeader.Height)
 	}
 
+	c.providerMutex.Lock()
 	if len(c.witnesses) > 0 {
 		if err := c.compareNewHeaderWithRandomWitness(newHeader); err != nil {
 			c.logger.Error("Error when comparing new header with one from a witness", "err", err)
```

Comment:
```suggestion
			c.providerMutex.Unlock()
			c.logger.Error("Error when comparing new header with one from a witness", "err", err)
```

otherwise mutex will stay locked after you `return err`